en/docs/security-announcements/cve-justifications/2025/CVE-2025-22874.md

```diff
@@ -0,0 +1,35 @@
+---
+title: CVE-2025-22874
+category: security-announcements
+date: June 19 2025
+---
+
+# CVE-2025-22874
+
+<p class="doc-info">WSO2 Products Impacted: No</p>
+<p class="doc-info">Customers Actions Required: No</p>
+---
+
+### REPORTED VULNERABILITY
+A flaw was found in Go's crypto/x509 package. This vulnerability allows improper certificate validation, bypassing policy constraints via using ExtKeyUsageAny in VerifyOptions.KeyUsages[^1].
+
+### REPORTED PRODUCTS
+* WSO2 Choreo Connect : 1.2.0
+
+### WSO2 JUSTIFICATION
+According to CVE considerations, the root cause lies in the `Verify` method within the `crypto/x509` package, which is part of the standard Go library. When calling the Verify function with a VerifyOptions.KeyUsages that contains ExtKeyUsageAny, policy validation is unintentionally disabled [^2].
+
+The vulnerability affects Go versions prior to 1.24.4. The v0.4.38 version of grpc-health-probe binary packaged in Choreo Connect, which is the latest available release of the tool at this time, is built with Go 1.24.2. Although the Go version has been updated to 1.24.4 in the master branch of the grpc-health-probe repository[^3], a new release containing this fix has not yet been published. As a result, it is currently not possible to incorporate a patched version of grpc-health-probe into the product.
```

Comment:
```suggestion
The vulnerability affects the aforementioned package in Go versions prior to 1.24.4. The v0.4.38 version of `grpc-health-probe` binary packaged in Choreo Connect, which is the latest available release of the tool at this time, is built with Go 1.24.2. Although the Go version has been updated to 1.24.4 in the master branch of the `grpc-health-probe` repository[^3], a new release containing the fixed `crypto/x509` has not yet been published. As a result, it is currently not possible to incorporate a patched version of `grpc-health-probe` into the product.
```

Reply:
Added in a916dd206dcf63ffd16258bd8d295b9658063bcc